File: ltt_server/src/model/tag.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
pub struct TagInfo{
    tagid:u32,
    tagname:String,
}

#[derive(Deserialize, Serialize)]
pub struct TagSet{
    tagsetname:String,
    tags:Vec<TagInfo>
}

#[derive(Deserialize, Serialize)]
pub struct AllTags{
    tagsets:Vec<TagSet>
}

File: ltt_client/src/store/models/tag.ts
export class TagInfo{
    constructor(
        public tag_id:number,
        public tag_name:string,
        public artcnt:number
    ) {
    }
}
export class TagSet{
    constructor(
        public tagsetname:string,
        public tags:TagInfo[]
    ) {
    }
}
export class AllTags{
    constructor(
        public tagsets:TagSet[],
    ) {
    }
    static pre(){
        return new AllTags([])
    }
}